keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ArtifactInterfaces.kt

```diff
@@ -0,0 +1,67 @@
+package com.netflix.spinnaker.keel.api
+
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactType
+
+/**
+ * Implemented by [ResourceSpec] or concrete resource types that (may) contain artifact information, typically compute
+ * resources.
+ *
+ * The fields on this interface are nullable because typically the spec will _not_ have that information available
+ * before the corresponding [ResourceHandler] has resolved the resource.
+ */
+interface ArtifactProvider {
```

Comment:
These new interfaces help identify which types of spec (may) contain artifact information that we can then associate with the resource in the database.